src/_pytest/nodes.py

```diff
@@ -484,38 +497,47 @@ def _check_initialpaths_for_relpath(session, fspath):
 
 
 class FSCollector(Collector):
-    def __init__(
-        self,
-        fspath: py.path.local,
-        parent=None,
-        config: Optional[Config] = None,
-        session: Optional["Session"] = None,
+    @classmethod
+    def from_parent(
+        cls,
+        parent: Node,
+        *,
+        fspath: Optional[py.path.local] = None,
+        fs_path: Optional[Path] = None,
         nodeid: Optional[str] = None,
-    ) -> None:
-        name = fspath.basename
-        if parent is not None:
+        name: Optional[str] = None,
+        **kw,
+    ):
+        """The public constructor."""
+        if fspath is not None:
+            assert fs_path is None
+            known_path = Path(fspath)
+        else:
+            assert fs_path is not None
+            known_path = fs_path
+            fspath = py.path.local(known_path)
+
+        if name is None:
+            name = known_path.name
             rel = fspath.relto(parent.fspath)
             if rel:
-                name = rel
-            name = name.replace(os.sep, SEP)
-        self.fspath = fspath
-
-        session = session or parent.session
+                name = str(rel)
```

Comment:
This will end up with a name containing `\` on Windows because we are no longer doing `name = name.replace(os.sep, SEP)`. Is that intentional?

Also `str` seems redudant:

```python
>>> local("src/pytest").relto(local("."))
'src\\pytest'
```

Reply:
relto is from py.path.local - relative pathlib paths are relative

Reply:
Ahh I see, thanks, but does the comment about `name.replace` still apply?

```python
>>> local("src/pytest").relto(local("."))
'src\\pytest'
```

Reply:
it does

Reply:
Ahh OK so this needs fixing then. 👍